src/mono/wasm/runtime/web-socket.ts

```diff
@@ -82,9 +87,16 @@ export function ws_wasm_create(uri: string, sub_protocols: string[] | null, rece
             setI32(<any>receive_status_ptr + 8, 1);// end_of_message: true
             receive_promise_control.resolve();
         });
+
+        // cleanup the delegate proxy
+        ws[wasm_ws_on_closed].dispose();
```

Comment:
is it guaranteed that this property will have a dispose method? i only see you assert that it's a function above

Reply:
Yes, it's mostly sanity check, because we are the only caller.